service/history/workflow/mutable_state_impl.go

```diff
@@ -8103,17 +8109,20 @@ func (ms *MutableStateImpl) StartDeploymentTransition(deployment *deploymentpb.D
 	// start because its directive deployment no longer matches workflows effective deployment.
 	// If the WFT is started but not finished, we let it run its course, once it's completed, failed
 	// or timed out a new one will be scheduled.
-	if ms.HasPendingWorkflowTask() && !ms.HasStartedWorkflowTask() &&
+	if ms.HasPendingWorkflowTask() && !ms.HasStartedWorkflowTask() {
 		// Speculative WFT is directly (without transfer task) added to matching when scheduled.
 		// It is protected by timeout on both normal and sticky task queues.
 		// If there is no poller for previous deployment, it will time out,
 		// and will be rescheduled as normal WFT.
-		ms.GetPendingWorkflowTask().Type != enumsspb.WORKFLOW_TASK_TYPE_SPECULATIVE {
-		// sticky queue was just cleared, so the following call only generates
-		// a WorkflowTask, not a WorkflowTaskTimeoutTask.
-		err := ms.taskGenerator.GenerateScheduleWorkflowTaskTasks(ms.GetPendingWorkflowTask().ScheduledEventID)
-		if err != nil {
-			return err
+		if ms.GetPendingWorkflowTask().Type != enumsspb.WORKFLOW_TASK_TYPE_SPECULATIVE {
+			// sticky queue was just cleared, so the following call only generates
+			// a WorkflowTask, not a WorkflowTaskTimeoutTask.
+			err := ms.taskGenerator.GenerateScheduleWorkflowTaskTasks(ms.GetPendingWorkflowTask().ScheduledEventID)
+			if err != nil {
+				return err
+			}
+		} else {
+			ms.logInfo("start transition did not reschedule pending speculative task")
```

Comment:
I think it would be nice to have the version string in this log line as well, but not critical